supervisor/archives_api.go

```diff
@@ -92,17 +92,25 @@ func (self ArchiveAPI) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			return
 		}
 
+		taskChan := make(chan string, 1)
 		// tell the supervisor to schedule a task
 		self.Tasks <- &db.Task{
-			Op:          db.RestoreOperation,
-			Owner:       params.Owner,
-			TargetUUID:  tid,
-			ArchiveUUID: id,
-			RestoreKey:  archive.StoreKey,
+			Op:           db.RestoreOperation,
+			Owner:        params.Owner,
+			TargetUUID:   tid,
+			ArchiveUUID:  id,
+			RestoreKey:   archive.StoreKey,
+			TaskUUIDChan: taskChan,
 		}
 
-		w.WriteHeader(200)
-		JSONLiteral(w, fmt.Sprintf(`{"ok":"scheduled"}`))
+		taskUUID := <-taskChan
+		if taskUUID == "" {
+			w.WriteHeader(500)
+			JSONLiteral(w, fmt.Sprintf(`{"ok":"error"}`))
```

Comment:
Why is this `{"ok":"error"}`? shouldn't we return something more like `{"error":"here is the reason for the error"}`?